ddtrace/context.py

```diff
@@ -114,16 +130,19 @@ def _set_current_span(self, span):
             self._parent_span_id = None
 
     def add_span(self, span):
+        # type: (Optional[Span]) -> None
```

Comment:
```suggestion
        # type: (Span) -> None
```

Reply:
Should `_set_current_span()` then accept only Span types then? The reason these optionals keep coming to bite us is due to `_set_current_span()` being called by both `add_span()` and `close_span()`

Reply:
`_set_current_span()` can accept `None` as the current span in a context can be `None` -- That shouldn't impact `add_span()`/`close_span()` (I would think)

Reply:
mypy is suddenly okay with this, should be good to fix following your suggestion!